distiller/fac_scraper/models.py

```diff
@@ -0,0 +1,57 @@
+"""
+Maintains references to Federal Audit Clearinghouse documents.
+"""
+
+import os
+
+from compositefk.fields import CompositeForeignKey
+from django.conf import settings
+from django.db import models
+
+from distiller.data.models import Audit
+
+
+class FacDocument(models.Model):
+    version = models.IntegerField()
+    report_id = models.CharField(max_length=8)
+    audit_year = models.DecimalField(
+        max_digits=4,
+        decimal_places=0,
+        help_text='Audit Year and DBKEY (database key) combined make up the primary key.'
+    )
+    dbkey = models.CharField(
+        max_length=6,
+        help_text='Audit Year and DBKEY (database key) combined make up the primary key.'
+    )
+    ein = models.CharField(
+        max_length=9,
+        help_text='Employer Identification Number'
+    )
+    fy_end_date = models.DateField()
+    fac_accepted_date = models.DateField()
+    date_received = models.DateField()
+    file_type = models.CharField(
+        max_length=8,
+        choices=(
+            ('form', 'form'),
+            ('audit', 'audit'),
+        )
+    )
+    file_name = models.CharField(max_length=32)  # <dbkey><audit-year><version>.<pdf | xlsx>
```

Comment:
Curious, can we construct the file name from all the information we have in this model instead? What's the advantage of storing the `file_name` explicitly? Wondering here because what happens if we have a divergence between `file_name` and its constituent parts?

Reply:
Yeah most likely we should be able to. At the moment, there is a `CloseSpider` exception thrown in the scraper if the filename doesn't match the expected one, to surface such a circumstance in a visible way. I figure it's very unlikely to happen, but since it's an assumption that's not proven, storing it instead seemed slightly safer.

Reply:
Sounds good to me, thanks for the explanation.